### ISSUE TITLE
Fullscreen borderless window does not exit on MacOS

### ISSUE THREAD

**Issue by odanek** [rust-windowing/winit#2283]
When window is created in the `Fullscreen::Borderless` mode setting `control_flow` to `ControloFlow::Exit` does not exit the event loop:
```rust
use winit::{
    event_loop::{ControlFlow, EventLoop},
    window::{Fullscreen, WindowBuilder},
};

fn main() {
    let event_loop = EventLoop::new();
    
    let window = WindowBuilder::new()
        .with_fullscreen(Some(Fullscreen::Borderless(None)))
        .build(&event_loop)
        .unwrap();

    event_loop.run(move |_, _, control_flow| {
        *control_flow = ControlFlow::Exit;
    });
}
```
Commenting out the `.with_fullscreen(...)` line does correctly exit the event loop.

Reproduced on:
`winit 0.26.1`
`rustc 1.60.0`
`MacOS 12.3.1`
`MacBook Pro M1`

**Comment by madsmtm** [comment]
Can't reproduce on macOS Mojave 10.14.6, but may have been fixed in https://github.com/rust-windowing/winit/pull/2292, could I get you to check?

**Comment by odanek** [comment]
@madsmtm I can confirm that the issue is fixed on master. It probably wasn't fixed by #2292 though because I wasn't able to reproduce the issue even using the commit before the #2292 merge.

**Comment by madsmtm** [comment]
It would be nice to know which commit fixed it, would you mind running a `git bisect` to figure out which one? If you can't be bothered, that's also fine!

**Comment by odanek** [comment]
@madsmtm I have checked again and it has been indeed fixed by #2292, commit 4c39b31. Sorry for the confusion.

**Comment by madsmtm** [comment]
Wonderful, thanks for your help!
